pkg/netpol/eval/resources.go

```diff
@@ -351,32 +323,27 @@ func (pe *PolicyEngine) getDisjointIPBlocks() ([]*k8s.IPBlock, error) {
 	return disjointRes, nil
 }
 
-func (pe *PolicyEngine) getServicePods(svcName, svcNamespace string) ([]*k8s.Pod, error) {
-	svc := pe.getServiceFromServiceNameAndNamespace(svcName, svcNamespace)
-	// todo: should return error if the service not found?
-	if svc == nil {
-		svcStr := types.NamespacedName{Namespace: svcNamespace, Name: svcName}
-		return nil, fmt.Errorf("service does not exist: %s", svcStr)
-	}
-	svcLabelsSelect, err := svc.ServicSelectorsAsLabelSelector()
+// GetSelectedPeers returns list of peers in the given namespace which match the given labels selector
+func (pe *PolicyEngine) GetSelectedPeers(selectors labels.Selector, namespace string) ([]Peer, error) {
+	res := make([]Peer, 0)
+	peers, err := pe.GetPeersList()
```

Comment:
instead of calling `pe.GetPeersList()` and then re-iterating the entire peers list in level of workloads, can do a simpler iteration as implemented in `GetPeersList` : iterate only `pe.podsMap` , and for a matching pod add it as a workload to the result.

Reply:
fix to avoid duplicate workloads in the result

Reply:
done